### PR TITLE
Add additional field for update_users_from_events 

### PR DESCRIPTION
**Pull request by mishranik**
_A summary of your pull request, including the what change you're making and why._

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/moengage/trackEvent/index.ts [hunk 1]
```diff
@@ -82,7 +82,14 @@ const action: ActionDefinition<Settings, Payload> = {
       default: {
         '@path': '$.properties'
       }
-    }
+    },    
+    update_existing_only: {
+      label: 'Update users from events',
+      type: 'boolean',
+      description: 'Settings to update the existing users through event sync',
```
Comment (tcgilbert) [inline packages/destination-actions/src/destinations/moengage/trackEvent/index.ts:89]: Can you confirm if I understand this field correctly? 

"When set to `true` events from segment will only update the user if the user already exists on your side"

Comment (mishranik) [reply packages/destination-actions/src/destinations/moengage/trackEvent/index.ts:89]: Yes, correct @tcgilbert

File: packages/destination-actions/src/destinations/moengage/trackEvent/index.ts [hunk 2]
```diff
@@ -82,7 +82,14 @@ const action: ActionDefinition<Settings, Payload> = {
...
+      required: false,
+      default: false
+    },
```
Comment (tcgilbert) [inline packages/destination-actions/src/destinations/moengage/trackEvent/index.ts:92]:
```suggestion
    update_existing_only: {
      label: 'Update Existing Users Only',
      type: 'boolean',
      description: 'If set to true, events from the segment will only trigger updates for users who already exist in Moengage.',
      required: false,
      default: false
    },
```

@mishranik could you update the label and description to the suggestion above, in order that the behavior is more clear?

**Comment by joe-ayoub-segment** [comment]
hi @mishranik from a coding perspective this change looks good. 

Have you tested this change in any way to ensure that it functionally works?

Kind regards,
Joe

**Comment by joe-ayoub-segment** [comment]
Hi @mishranik , looks like the unit tests are failing. Should be simple to fix. 
image

Could you also get the latest code from the Segment main branch please? I can see some other failures which hopefully will get fixed if you merge in the latest code. 

Best regards,
Joe

**Comment by mishranik** [comment]
@joe-ayoub-segment I tried pulling from Segment main branch and it is showing up do date, rest i have tired fixing the test case. Let me know if something is still missing

**Comment by joe-ayoub-segment** [comment]
Hi @mishranik this snapshot is still failing: 

image

**Comment by mishranik** [comment]
@joe-ayoub-segment my bad, failed to push the last update in the file.

**Comment by mishranik** [comment]
@joe-ayoub-segment can you please help me understand why exactly is last test case failed?

**Comment by joe-ayoub-segment** [comment]
hi @mishranik I'll rerun the unit tests now. We should be able to see if there's anything else failing. 
I do suspect that someone from the Segment team checked in some bad code for another Integration which is breaking the unit tests, so I'll be following up on that.

**Comment by joe-ayoub-segment** [comment]
Hi @mishranik , this snapshot is still failing. 
image


Also there are some issues which the team needs to fix here. Once these have been committed to our main branch I'll ask you to pull the code from there. But for now I think you only need to fix that failing test above.

**Comment by mishranik** [comment]
thanks @joe-ayoub-segment  for the update, I need some help in understanding what exactly is required in the snapshot test case, unable to understand why exactly it is failing

**Comment by joe-ayoub-segment** [comment]
Hi @mishranik , 

All you need to do is update the snapshots and then commit the changes. You can do this with the following command: 

`yarn jest --testPathPattern='./packages/destination-actions/src/destinations/moengage' --updateSnapshot`

Best regards,
Joe

**Comment by mishranik** [comment]
Thanks @joe-ayoub-segment  I have ran that command and pushed my changes

**Comment by joe-ayoub-segment** [comment]
Thanks @mishranik . We also fixed some of the issues on our side. Any chance you could get the latest from our main branch please?

Apologies for all the hassle!

**Comment by mishranik** [comment]
@joe-ayoub-segment pushed the latest pull from segment main branch

**Comment by joe-ayoub-segment** [comment]
Hi @mishranik - That's OK the reason the Validate check is failing is something I can fix pre-deploy. 
I'll merge and deploy the PR as it is, so no need for you to do anything now.

**Comment by joe-ayoub-segment** [comment]
hi @mishranik this PR has been deployed. Please confirm that the change works as expected!

**Pull request merged**